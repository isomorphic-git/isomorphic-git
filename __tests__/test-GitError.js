/* eslint-env node, browser, jasmine */
const snapshots = require('./__snapshots__/test-GitError.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')

const { models } = require('isomorphic-git/internal-apis')
const { E, GitError } = models

describe('GitError', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })
  it('creates an Error', async () => {
    let e = null
    try {
      throw new GitError(E.FileReadError, { filepath: 'foobar.txt' })
    } catch (err) {
      e = err
    }
    expect(e).not.toBeNull()
    expect(e.code).toBe(E.FileReadError)
    expect(e instanceof Error).toBe(true)
    expect(e instanceof GitError).toBe(true)
    expect(new Error() instanceof Error).toBe(true)
    expect(new Error() instanceof GitError).toBe(false)
    expect(e.toJSON()).toMatchSnapshot()
  })
})
