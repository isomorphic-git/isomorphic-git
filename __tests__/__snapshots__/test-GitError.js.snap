// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitError creates an Error 1`] = `
Object {
  "caller": undefined,
  "code": "FileReadError",
  "data": Object {
    "filepath": "foobar.txt",
  },
  "message": "Could not read file \\"foobar.txt\\".",
}
`;
